.github/workflows/main.yml

```diff
@@ -11,6 +11,19 @@ on:
       - '**'
 
 jobs:
+  macos:
+    runs-on: macOS-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Build the dependencies
+      run: |
+        make deps
+    - name: Build the app
+      run: |
+        make
```

Comment:
In order to keep the code-signing config and `ENABLE_HARDENED_RUNTIME` in the project, we could improve the Makefile:
- If it's Github Action (`GITHUB_ACTION` env is visible), we could disable code-signing in `xcodebuild_command`
- If not, just run as normal, so we we mini-bot still works as usual.

For instance,
```makefile
ifdef GITHUB_ACTION
xcodebuild_command=xcodebuild \
				  -scheme TogglDesktop \
				  -workspace src/ui/osx/TogglDesktop.xcworkspace  \
				  -configuration Debug CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO
else
xcodebuild_command=xcodebuild \
				  -scheme TogglDesktop \
				  -workspace src/ui/osx/TogglDesktop.xcworkspace  \
				  -configuration Debug
endif
```
Ref: https://help.github.com/en/github/automating-your-workflow-with-github-actions/virtual-environments-for-github-actions#default-environment-variables

Reply:
This doesn't solve anything though because we need notarization and signing in GitHub actions too.

Reply:
Unfortunately, If we directly disable the DEVELOPMENT_TEAM and ENABLE_HARDENED_RUNTIME in the Xcode Project, I couldn't build the app in the local machine anymore, even the buildbot is also affected.

In general, my suggestion would be
- Keep the DEVELOPMENT_TEAM and ENABLE_HARDENED_RUNTIME in Xcode Project
- Improve the makefile in `default` route: Disable codesigning in xcodebuild if it's Github action => So the Github action could build for testing without dealing with the codesign.
- For the Notarization & Codesign in Github action: We could improve later in `codesign.sh` and `notarize.sh` from the branding repo.

## Result
- Able to build in local machine
- Able to run for testing in Github Action (No Codesiging)
- Able to do notarization + Codesign when we support in the Branding repo.

Please correct me if I'm wrong, or please share us your proposal 🌮